File: leetcode/633.swift
class Solution {
    func isSquareRootInteger(_ num: Int) -> Bool {
        let sqrtVal = sqrt(Double(num))
        return sqrtVal.truncatingRemainder(dividingBy: 1) == 0
    }

    func judgeSquareSum(_ c: Int) -> Bool {
        var left = 0
        var right = Int(sqrt(Double(c)))

        while left <= right {
            let sum = left * left + right * right
            if sum == c {
                return true
            } else if sum < c {
                left += 1
            } else {
                right -= 1
            }
        }

        return false
    }
}

File: leetcode/633.java
class Solution {
    public boolean judgeSquareSum(int c) {
       for(int i = 0; i <= Math.sqrt(c)+1; i++) {
        int diff = c - (i * i);
        int sqrtdiff = (int) Math.sqrt(diff);
        if (sqrtdiff*sqrtdiff == diff) {
            return true;
        }
       } 
       return false;
    }
}
